tensorflow_lite_support/python/test/task/audio/core/tensor_audio_test.py

```diff
@@ -0,0 +1,170 @@
+    # Fails loading audio data from a NumPy array with an input size
+    # matching sample rate.
+    with self.assertRaisesRegex(
+        ValueError,
+        f"Input audio contains an invalid number of samples. "
+        f"Expect {_BUFFER_SIZE}."):
+      array = np.random.rand(_SAMPLE_RATE, _CHANNELS).astype(np.float32)
+      self.test_tensor_audio.load_from_array(array)
+
+  def test_load_from_array_fails_with_input_size_less_than_sample_rate(self):
+    # Fails loading audio data from a NumPy array with an input size
+    # less than sample rate.
+    with self.assertRaisesRegex(
+        ValueError,
+        f"Input audio contains an invalid number of samples. "
+        f"Expect {_BUFFER_SIZE}."):
+      input_buffer_size = 10000
+      self.assertLess(input_buffer_size, _SAMPLE_RATE)
+      array = np.random.rand(input_buffer_size, _CHANNELS).astype(np.float32)
+      self.test_tensor_audio.load_from_array(array)
+
+  def test_load_from_array_fails_with_invalid_number_of_channels(self):
+    # Fails loading audio data from a NumPy array with an invalid
+    # number of input channels.
+    with self.assertRaisesRegex(
+        ValueError,
+        f"Input audio contains an invalid number of channels. "
+        f"Expect {_CHANNELS}."):
+      array = np.random.rand(_BUFFER_SIZE, 2).astype(np.float32)
+      self.test_tensor_audio.load_from_array(array)
+
+  def test_load_from_array_fails_with_too_many_input_samples(self):
+    # Fails loading audio data from a NumPy array with a sample count
+    # exceeding TensorAudio's internal buffer capacity.
+    with self.assertRaisesRegex(
+        ValueError,
+        f"Input audio contains an invalid number of samples. "
+        f"Expect {_BUFFER_SIZE}."):
+      array = np.random.rand(20000, _CHANNELS).astype(np.float32)
+      self.test_tensor_audio.load_from_array(array)
+
+  @mock.patch("sounddevice.InputStream", return_value=mock.MagicMock())
+  def test_load_from_audio_record(self, mock_input_stream):
+    record = audio_record.AudioRecord(_CHANNELS, _SAMPLE_RATE, _BUFFER_SIZE)
+
+    # Get AudioRecord's audio callback function.
+    _, mock_input_stream_init_args = mock_input_stream.call_args
+    callback_fn = mock_input_stream_init_args["callback"]
+
+    # Create dummy data to feed to the AudioRecord instance.
+    chunk_size = int(_BUFFER_SIZE * 0.5)
+    input_data = []
+    for _ in range(3):
+      dummy_data = np.random.rand(chunk_size, _CHANNELS).astype(float)
+      input_data.append(dummy_data)
+      callback_fn(dummy_data)
+    expected_data = np.concatenate(input_data[-2:])
+
+    # Load audio data into TensorAudio from the AudioRecord instance.
+    self.test_tensor_audio.load_from_audio_record(record)
+
+    # Assert read all data in the float buffer.
+    testing.assert_almost_equal(self.test_tensor_audio.buffer, expected_data)
+
+  def test_load_from_audio_record_fails_with_invalid_buffer_size(self):
```

Comment:
Sounddevice doesn't run on devices without a soundcard so you'll need to mock InputStream on all test cases that use AudioRecord.

Reply:
@khanhlvg Alright, will fix it